spyrmsd/graph.py

```diff
@@ -1,27 +1,98 @@
+    from spyrmsd.graphs.gt import match_graphs as gt_match_graphs
+    from spyrmsd.graphs.gt import num_edges as gt_num_edges
+    from spyrmsd.graphs.gt import num_vertices as gt_num_vertices
+    from spyrmsd.graphs.gt import vertex_property as gt_vertex_property
 
+    _available_backends.append("graph-tool")
 except ImportError:
-    try:
-        from spyrmsd.graphs.nx import (
-            cycle,
-            graph_from_adjacency_matrix,
-            lattice,
-            match_graphs,
-            num_edges,
-            num_vertices,
-            vertex_property,
-        )
-    except ImportError:
-        raise ImportError("graph_tool or NetworkX libraries not found.")
+    warnings.warn("The graph-tool backend does not seem to be installed.", stacklevel=2)
+
+try:
+    from spyrmsd.graphs.nx import cycle as nx_cycle
+    from spyrmsd.graphs.nx import (
+        graph_from_adjacency_matrix as nx_graph_from_adjacency_matrix,
+    )
+    from spyrmsd.graphs.nx import lattice as nx_lattice
+    from spyrmsd.graphs.nx import match_graphs as nx_match_graphs
+    from spyrmsd.graphs.nx import num_edges as nx_num_edges
+    from spyrmsd.graphs.nx import num_vertices as nx_num_vertices
+    from spyrmsd.graphs.nx import vertex_property as nx_vertex_property
+
+    _available_backends.append("networkx")
+except ImportError:
+    warnings.warn("The networkx backend does not seem to be installed.", stacklevel=2)
+
+
+def _validate_backend(backend):
+    standardized_backend = _alias_to_backend.get(backend.lower())
+    if standardized_backend is None:
+        raise ValueError(f"The {backend} backend is not recognized or supported")
+    if standardized_backend not in _available_backends:
+        raise ImportError(f"The {backend} backend doesn't seem to be installed")
+    return standardized_backend
+
+
+def available_backends():
+    return _available_backends
+
+
+def set_backend(backend):
+    global _current_backend
+    backend = _validate_backend(backend)
+
+    ## Check if we actually need to switch backends
+    if backend == _current_backend:
+        warnings.warn(f"The backend is already {backend}", stacklevel=2)
+        return
+
+    global cycle, graph_from_adjacency_matrix, lattice, match_graphs, num_edges, num_vertices, vertex_property
+
+    if backend == "graph-tool":
+        cycle = gt_cycle
+        graph_from_adjacency_matrix = gt_graph_from_adjacency_matrix
+        lattice = gt_lattice
+        match_graphs = gt_match_graphs
+        num_edges = gt_num_edges
+        num_vertices = gt_num_vertices
+        vertex_property = gt_vertex_property
+
+    elif backend == "networkx":
+        cycle = nx_cycle
+        graph_from_adjacency_matrix = nx_graph_from_adjacency_matrix
+        lattice = nx_lattice
+        match_graphs = nx_match_graphs
+        num_edges = nx_num_edges
+        num_vertices = nx_num_vertices
+        vertex_property = nx_vertex_property
+
+    _current_backend = backend
+
 
 __all__ = [
```

Comment:
Please remove this again, I was wrong. Sorry.

Reply:
Don't worry, I wasn't sure about this either. I removed it again!